app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -150,8 +153,44 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
 
     @Override
     public void onLongPress(MotionEvent e) {
-        // Also switch on long-press for accessibility
-        setMainImage(true, true);
+        openCurrentMainImageInGallery();
+    }
+
+    private void openCurrentMainImageInGallery() {
+        ImageType wantedImageType = imageTypes.get(mainImageIndex);
+
+        File file = null;
+
+        switch (wantedImageType) {
+            case IMAGE_FRONT:
+                file = Utils.retrieveCardImageAsFile(this, loyaltyCardId, ImageLocationType.front);
+                break;
+            case IMAGE_BACK:
+                file = Utils.retrieveCardImageAsFile(this, loyaltyCardId, ImageLocationType.back);
+                break;
+            case BARCODE:
+                Toast.makeText(this, R.string.barcodeLongPressMessage, Toast.LENGTH_SHORT).show();
+                break;
```

Comment:
If we return here instead of break...

```suggestion
                return;
```